tests/test_paper.py

```diff
@@ -0,0 +1,164 @@
+import trio
+import pytest
+import tractor
+import math
+import os
+from shutil import rmtree
+from piker.config import get_app_dir
+from piker.log import get_logger
+from piker.clearing._messages import ( 
+    Order                                    
+)
+from uuid import uuid4
+from typing import (
+    AsyncContextManager,
+    Literal,
+)
+from functools import partial
+from piker.pp import (
+    open_trade_ledger,
+    open_pps,
+)
+from piker.clearing import (
+    open_ems,
+)
+from piker.clearing._client import (
+    OrderBook,
+)
+from piker.clearing._messages import (
+    BrokerdPosition
+)
+from exceptiongroup import BaseExceptionGroup
+
+log = get_logger(__name__)
+
+@pytest.fixture(scope="session")
+def paper_cleanup():
+    yield
+    app_dir = get_app_dir('piker')
+    if os.path.isfile(app_dir):       
+        rmtree(app_dir)
+        assert not os.path.isfile(app_dir)
+
+def test_paper_trade(
+    open_test_pikerd: AsyncContextManager,
+    paper_cleanup: None
+):
+
+    cleared_price: float
+    test_exec_mode='live'
+    test_account = 'paper'
+    test_size = 1 
+    test_broker = 'kraken'
+    test_brokers = [test_broker]
+    test_symbol = 'xbtusdt' 
+    test_fqsn = f'{test_symbol}.{test_broker}'
+    test_pp_account = 'piker-paper'
+    positions: dict[
+        # brokername, acctid
+        tuple[str, str],
+        list[BrokerdPosition],
+    ]
+
+    async def _async_main(
+        open_pikerd: AsyncContextManager,
+        action: Literal['buy', 'sell'] | None = None,
+        price: int = 30000,
+        assert_entries: bool = False,
+    ) -> None:
+        """Spawn a paper piper actor, place a trade and assert entries are present
+        in both trade ledger and pps tomls. Then restart piker and ensure 
+        that pps from previous trade exists in the ems pps. 
+        Finally close the position and ensure that the position in pps.toml is closed. 
+        """
+
+        oid: str = str(uuid4())
+        book: OrderBook
+        global cleared_price
+        global positions
+        
+        # Set up piker and EMS
+        async with (
+            open_pikerd() as (_, _, _, services),
+            open_ems(
+                test_fqsn,
+                mode=test_account,
+            ) as (
+                book,
+                trades_stream,
+                pps,
+                accounts,
+                dialogs,
+            ),
+        ):
+
+            # Send order to EMS
+            if action: 
+                order = Order(
+                    exec_mode=test_exec_mode,
+                    action=action,
+                    oid=oid,
+                    account=test_account,
+                    size=test_size,
+                    symbol=test_fqsn,
+                    price=price,
+                    brokers=test_brokers
+                )
+
+                book.send(order)
```

Comment:
put a comment about this being a sync call to an async queue maybe just so that peeps who come to learn from this test know what the heck is going on 😉